app/models/application.rb

```diff
@@ -0,0 +1,28 @@
+class Application < ApplicationRecord
+  has_many :application_pets
+  has_many :pets, through: :application_pets
+  has_many :application_pets
+  has_many :pets, through: :application_pets
+
+  validates :name, presence: true
+  validates :street_address, presence: true
+  validates :city, presence: true
+  validates :state, presence: true
+  validates :zip_code, presence: true
+  validates :description, presence: true
+  validates :status, presence: true
+
+  def address_join
+    "#{street_address}, #{city}, #{state}, #{zip_code}"
+  end
+
+  def change_status
+    check_values = application_pets.map{|pet| pet.status}
```

Comment:
It's okay to use Ruby here because the ActiveRecord equivalent here is pretty hard and not necessarily more elegant!

I'm curious if the nil check here will cause applications to be prematurely considered "Rejected" if none of their application_pets have been approved. The status column is probably defaulting to nil, that's why I'm wondering! Is the nil check necessary? THere is probably a nil edge case I'm not considering